official/resnet/keras/resnet_model.py

```diff
@@ -69,8 +69,10 @@ def identity_block(input_tensor, kernel_size, filters, stage, block):
                     kernel_regularizer=regularizers.l2(L2_WEIGHT_DECAY),
                     name=conv_name_base + '2a')(input_tensor)
   x = layers.BatchNormalization(axis=bn_axis,
+                                scale=False,
                                 momentum=BATCH_NORM_DECAY,
                                 epsilon=BATCH_NORM_EPSILON,
+                                fused=True,
```

Comment:
fused is default True.  I see below it gets confusing. My concern with setting it is that people then think you need to worry about this when by default it is True.  I am ok with it, but I wanted to understand and a little digging made me more confused. Even if you say `True` it will still treat it as if you set it to none because it has to pass `_raise_if_fused_cannot_be_used`

The document says

> fused: if None or True, use a faster, fused implementation if possible. If False, use the system recommended implementation.

I guess why is this now different in V2. I think this comment should be be higher up to indicate it still "may" not use fused if the input rank is not 4.
```python
if self._USE_V2_BEHAVIOR:
      if fused:
        self._raise_if_fused_cannot_be_used()
      # We leave fused as None if self._fused_can_be_used()==True, since we
      # still may set it to False in self.build() if the input rank is not 4.
      elif fused is None and not self._fused_can_be_used():
        fused = False
    elif fused is None:
      fused = True
    self.supports_masking = True

```

Reply:
I see. I knew that the underlying op is `FusedBatchNorm` by default but this logic here successfully confused me :) Will get the `fused=True` removed.

Reply:
I am still confused but for V2 it looks like even if you set to True it can still flip back to false, but if you Fused=True then instead of silently it should raise an error.  I say should as looking at self.build I am not totally sure if it does not "secretly" flip it to false.  I also don't get why V2 is different unless virtual batch is a new thing.  If removing fused gets you the desired result we should drop it.